src/index.ts

```diff
@@ -0,0 +1,436 @@
+import {
+    DerEncodedPublicKey,
+    HttpAgentRequest,
+    PublicKey as PublicKeyInterface,
+    requestIdOf,
+    Signature,
+    SignIdentity,
+} from '@dfinity/agent';
+import {DelegationChain} from '@dfinity/identity';
+import {Principal} from '@dfinity/principal';
+import {Buffer} from 'buffer';
+
+window.Buffer = Buffer;
+const domainSeparator = Buffer.from(new TextEncoder().encode('\x0Aic-request'));
+let globalStoicOrigin = 'https://www.stoicwallet.com';
+
+class PublicKey implements PublicKeyInterface {
+    constructor(private readonly _der: Uint8Array, private readonly _type: string) {}
+
+    public getType() {
+        return this._type;
+    }
+    public toDer(): DerEncodedPublicKey {
+        return Object.assign(this._der, {__derEncodedPublicKey__: undefined});
+    }
+}
+
+export class StoicIdentity extends SignIdentity {
+    constructor(
+        protected override readonly _principal: Principal,
+        private readonly _publicKey: PublicKey,
+    ) {
+        super();
+    }
+
+    static disconnect() {
+        return stoicLogout();
+    }
+
+    /**
+     * @param {string} host Override the wallet host to connect to. This will be preserved across
+     *   calls to this or other methods that accept a host input. Defaults to
+     *   https://www.stoicwallet.com if host has not been a truthy string yet.
+     * @returns StoicIdentity
+     */
+    static async connect(host?: string): Promise<StoicIdentity> {
+        if (host) {
+            globalStoicOrigin = host;
+        }
+        const data = await stoicLogin(globalStoicOrigin);
+        return new StoicIdentity(
+            Principal.fromText(data.principal),
+            new PublicKey(data.key, data.type),
+        );
+    }
+
+    /**
+     * @param {string} host Override the wallet host to connect to. This will be preserved across
+     *   calls to this or other methods that accept a host input. Defaults to
+     *   https://www.stoicwallet.com if host has not been a truthy string yet.
+     * @param {boolean} ignoreAccounts Optional, defaults to false. If true, accounts will not be
+     *   loaded as part of the authorization process.
+     * @returns StoicIdentity | undefined
+     */
+    static async load(
+        host?: string,
+        ignoreAccounts: boolean = false,
```

Comment:
Added this optional input because the `stoicIdentity.accounts` call [from version `1.5.0`](https://github.com/Toniq-Labs/stoic-identity/commit/62142582b8de5af55ac358a0d46978547b444b27#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556) is causing errors in my browser. Specifically, this error:

```
[Error] ReferenceError: Can't find variable: crossOriginIsolated
	Global Code (www.stoicwallet.com:1:616)
```

My project works fine without the `.accounts()` call.